packages/amplify-cli/src/extensions/amplify-helpers/update-backend-config.js

```diff
@@ -17,21 +18,11 @@ function updateBackendConfigAfterResourceAdd(category, resourceName, options) {
   }
 }
 
-function updateBackendConfigDependsOn(category, resourceName, attribute, value) {
+function updateBackendConfigAfterResourceUpdate(category, resourceName, attribute, value) {
   const backendConfigFilePath = pathManager.getBackendConfigFilePath();
   const backendConfig = getExistingBackendConfig(backendConfigFilePath);
-
-  if (!backendConfig[category]) {
-    backendConfig[category] = {};
-    backendConfig[category][resourceName] = {};
-  } else if (!backendConfig[category][resourceName]) {
-    backendConfig[category][resourceName] = {};
-  }
-  backendConfig[category][resourceName][attribute] = value;
-
-  const jsonString = JSON.stringify(backendConfig, null, 4);
-
-  fs.writeFileSync(backendConfigFilePath, jsonString, 'utf8');
+  _.set(backendConfig, [category, resourceName, attribute], value);
```

Comment:
Does this set the values to empty object {} if the corresponding backendConfig[category]  or backendConfig[category][resourceName] is not present?

Reply:
yes it will create the nested objects if they don't exist